### PR TITLE
feat: refactor Assertion/Attestation to Jackson ObjectMapper

### LINKED ISSUE #8907: ProjectPasskeys: Refactor Assertion/Attestation to Jackson ObjectMapper
Currently all options/init requests, option/init responses, and assertion/attestation results are managed as Jackson `JsonNode`, which basically means that all options are managed by hand.

To address it, I propose factory approach with Jackson Object mapping:

```java
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.io.IOException;

public class MyFactoryClass {
    // Example fields
    private String name;
    private int age;

    // Default constructor
    public MyFactoryClass() {
    }

    // Constructor using fields
    public MyFactoryClass(String name, int age) {
        this.name = name;
        this.age = age;
    }

    // Getters and setters
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    // Method to convert JsonNode to MyFactoryClass instance
    public static MyFactoryClass fromJsonNode(JsonNode jsonNode) throws IOException {
        ObjectMapper objectMapper = new ObjectMapper();
        // Convert JsonNode to MyFactoryClass
        return objectMapper.treeToValue(jsonNode, MyFactoryClass.class);
    }

    public static void main(String[] args) {
        ObjectMapper objectMapper = new ObjectMapper();
        String json = "{\"name\":\"John Doe\",\"age\":30}";

        try {
            // Parse JSON string to JsonNode
            JsonNode jsonNode = objectMapper.readTree(json);
            // Convert JsonNode to MyFactoryClass instance
            MyFactoryClass myFactoryClass = MyFactoryClass.fromJsonNode(jsonNode);
            System.out.println("Name: " + myFactoryClass.getName() + ", Age: " + myFactoryClass.getAge());
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
```

https://stackoverflow.com/questions/6311921/best-way-to-use-jackson-jsonnodefactory

This would simplify management of the requests, and let Jans to have class method for validating requests/responses.

### PR DESCRIPTION
**Pull request by duttarnab**
closes #8907

### CODE REVIEW

File: jans-fido2/model/src/main/java/io/jans/fido2/model/common/PublicKeyCredentialParameters.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package io.jans.fido2.model.common;
+
+public class PublicKeyCredentialParameters {
+    private int alg;
+    private String type = PublicKeyCredentialType.PUBLIC_KEY.getKeyName();
+
+    public PublicKeyCredentialParameters() {
+    }
+
+    public PublicKeyCredentialParameters(int alg, String type) {
```
Comment (yackermann) [inline jans-fido2/model/src/main/java/io/jans/fido2/model/common/PublicKeyCredentialParameters.java]: type can be set by default to "public-key"

File: jans-fido2/model/src/main/java/io/jans/fido2/model/common/PublicKeyCredentialDescriptor.java [hunk 2]
```diff
@@ -20,15 +22,15 @@
...
+    private String[] transports;
     private String id;
 
     public PublicKeyCredentialDescriptor(String type, String id) {
         this.type = type;
         this.id = id;
     }
 
-    public PublicKeyCredentialDescriptor(String type, String transports[], String id) {
+    public PublicKeyCredentialDescriptor(String type, String[] transports, String id) {
```
Comment (yackermann) [inline jans-fido2/model/src/main/java/io/jans/fido2/model/common/PublicKeyCredentialDescriptor.java]: Type can be set to default "public-key"

File: jans-fido2/model/src/main/java/io/jans/fido2/model/assertion/AssertionResult.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    public Response getResponse() {
+        return response;
+    }
+
+    public void setResponse(Response response) {
+        this.response = response;
+    }
+
+    public static AssertionResult createAssertionResult(String id, String type, String rawId, Response response) {
```
Comment (yackermann) [inline jans-fido2/model/src/main/java/io/jans/fido2/model/assertion/AssertionResult.java]: Type can be default "public-key"

File: jans-fido2/model/src/main/java/io/jans/fido2/model/assertion/AssertionResult.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+public class AssertionResult extends SuperGluuSupport {
+    private String id;
+    private String type;
+    private String rawId;
+    private Response response;
+
+    public AssertionResult() {
+    }
+
+    public AssertionResult(String id, String type, String rawId, Response response) {
```
Comment (yackermann) [inline jans-fido2/model/src/main/java/io/jans/fido2/model/assertion/AssertionResult.java]: Type can be default "public-key"

File: jans-fido2/server/src/main/java/io/jans/fido2/service/sg/converter/AssertionSuperGluuController.java [hunk 5]
```diff
@@ -176,34 +175,31 @@ public ObjectNode buildFido2AssertionStartResponse(String userName, String keyHa
...
-
-        // Manadatory parameter
-        params.put("type", "public-key");
-
-        params.put("id", authenticateResponse.getKeyHandle());
-
-        params.put("rawId", authenticateResponseString);
+        assertionResult.setSuperGluuRequest(true);
+        assertionResult.setSuperGluuRequestMode(oneStep ? SuperGluuMode.ONE_STEP.getMode() : SuperGluuMode.TWO_STEP.getMode());
+        assertionResult.setType("public-key");
```
Comment (yackermann) [inline jans-fido2/server/src/main/java/io/jans/fido2/service/sg/converter/AssertionSuperGluuController.java]: Constructor should have type set by default.

If you actually want to set type, then it should use instance of PublicKeyCredentialType

**Review by yackermann — CHANGES_REQUESTED** [review]

**Comment by yackermann** [comment]
@duttarnab overall good, but remove method `type` argument from everywhere, and just set it to "public-key" via PublicKeyCredentialType.

File: jans-fido2/model/src/main/java/io/jans/fido2/model/common/PublicKeyCredentialParameters.java (hunk 1 above)
Comment (duttarnab) [reply jans-fido2/model/src/main/java/io/jans/fido2/model/common/PublicKeyCredentialParameters.java]: done

File: jans-fido2/model/src/main/java/io/jans/fido2/model/common/PublicKeyCredentialDescriptor.java (hunk 2 above)
Comment (duttarnab) [reply jans-fido2/model/src/main/java/io/jans/fido2/model/common/PublicKeyCredentialDescriptor.java]: done

File: jans-fido2/model/src/main/java/io/jans/fido2/model/assertion/AssertionResult.java (hunk 3 above)
Comment (duttarnab) [reply jans-fido2/model/src/main/java/io/jans/fido2/model/assertion/AssertionResult.java]: done

File: jans-fido2/model/src/main/java/io/jans/fido2/model/assertion/AssertionResult.java (hunk 4 above)
Comment (duttarnab) [reply jans-fido2/model/src/main/java/io/jans/fido2/model/assertion/AssertionResult.java]: done

File: jans-fido2/server/src/main/java/io/jans/fido2/service/sg/converter/AssertionSuperGluuController.java (hunk 5 above)
Comment (duttarnab) [reply jans-fido2/server/src/main/java/io/jans/fido2/service/sg/converter/AssertionSuperGluuController.java]: done

**Comment by duttarnab** [comment]
> @duttarnab overall good, but remove method type argument from everywhere, and just set it to "public-key" via PublicKeyCredentialType.

done

**Pull request merged**